pika/connection.py

```diff
@@ -786,6 +810,10 @@ def channel(self, on_open_callback, channel_number=None):
         :rtype: pika.channel.Channel
 
         """
+        if not self.is_open:
+            raise exceptions.ConnectionClosed(
```

Comment:
REVIEWME do not allow channel to be created on connection that hasn't completed transition to OPEN state to avoid race condition with connection setup.